src/jit/lower.cpp

```diff
@@ -381,8 +383,44 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
         }
         break;
     case GT_CALL:
-        NYI("Call with TYP_LONG return value");
-        break;
+        {
+            GenTree* parent = data->parent;
+            if (parent != nullptr)
+            {
+                if (parent->gtOper == GT_STORE_LCL_VAR)
+                {
+                    // If parent is already a STORE_LCL_VAR, we can skip it if
+                    // it is already marked as lvIsMultiRegArgOrRet
+                    unsigned varNum = parent->AsLclVarCommon()->gtLclNum;
+                    if (comp->lvaTable[varNum].lvIsMultiRegArgOrRet)
+                    {
+                        break;
+                    }
+                    else if (!comp->lvaTable[varNum].lvPromoted)
+                    {
+                        // If var wasn't promoted, we can just set lvIsMultiRegArgOrRet
+                        comp->lvaTable[varNum].lvIsMultiRegArgOrRet = true;
+                        break;
+                    }
+                }
+
+                // Otherwise, we need to force var = call()
+                GenTreePtr* treePtr = nullptr;
+                parent = tree->gtGetParent(&treePtr);
+
+                GenTreeStmt* asgStmt = comp->fgInsertEmbeddedFormTemp(treePtr);
```

Comment:
@CarolEidt can new asgStmt become a top-level stmt since the check above is "if (parent !=nullptr)" that is GT_CALL node has a parent.  The case where GT_CALL node is a top-level stmt without any parent, there is no need to force IR to be var=call.  Did I mis-understood anything?